src/main/kotlin/com/amazon/ion/plugin/intellij/Ion.flex

```diff
@@ -0,0 +1,309 @@
+package com.amazon.ion.plugin.intellij;
+
+import com.intellij.psi.tree.IElementType;
+import com.amazon.ion.plugin.intellij.psi.IonTypes;
+import com.intellij.lexer.FlexLexer;
+import com.intellij.psi.TokenType;
+
+import static com.amazon.ion.plugin.intellij.helpers.ContentCorrectnessHelper.*;
+
+%%
+
+%class IonLexer
+%public
+%implements FlexLexer
+%unicode
+%function advance
+%type IElementType
+%eof{  return;
+%eof}
+
+%{
+    private java.util.Stack<Integer> _zzStateStack = new java.util.Stack<Integer>();
+%}
+/* Whitespace */
+LineTerminator = \r|\n|\r\n
+WhiteSpace     = {LineTerminator} | [ \t\f]
+
+/* comments */
+Comment = {TraditionalComment} | {EndOfLineComment} | {DocumentationComment}
+InputCharacter = [^\r\n]
+TraditionalComment   = "/*" [^*] ~"*/" | "/*" "*"+ "/"
+EndOfLineComment     = "//" {InputCharacter}*
+DocumentationComment = "/**" {CommentContent} "*"+ "/"
+CommentContent       = ( [^*] | \*+ [^/*] )*
+
+/* Global */
+Identifier = ([A-Za-z_$]) ([0-9A-Za-z_$])*
+Comma = ","
+LBrace = "{"
+RBrace = "}"
+LBracket = "["
+RBracket = "]"
+LParen = "("
+RParen = ")"
+
+/* Key */
+KeyCharacter = [^:\ \n\r\t\f\\\'\"]
+
+/* Operators */
```

Comment:
This was existing in the [Ion flex description](https://github.com/amzn/ion-intellij-plugin/blob/master/src/com/amazon/ion/plugin/intellij/Ion.flex#L60). I did not modify this. Before I change this, I need to do more testing to make sure it doesn't cause any other regressions. Since this was existing, I recommend just leaving it how it is and then we can open an issue or future PR to fix this specifically.

Reply:
After reading the spec excerpt above in more detail, I think your statement regarding each of the double values (`++`, `--`, `&&`, etc) does not follow the excerpt actually, specifically this statement:

---

 **An operator** is an **unquoted** sequence of one **or more** of the following nineteen ASCII characters: !#%&*+-./;<=>?@^`|~

---

In my interpretation of this statement, this means that "An [one] operator .... is an unquoted sequence of one or more ...." of the special operator characters. So this means the parser should interpret a sequence of operator characters as a single operator, i.e. `+=`, `|=`, `&&&&`, etc are all a single operator according to that statement.

According to your statement about double values (`++`, etc.) the following example from the excerpt should have 10 symbols but it in fact has seven symbols according to the spec:

> (a==b&&c==d)     // S-expression with seven symbols

Reply:
I agree with your interpretation, that's my mistake!

What drew my attention to this was the listing of "recognized" operators. Ion doesn't consider `++` to be more special than `+++` or `++++`, so having different logic for certain operators felt a bit off.